tiledb/sm/c_api/tiledb.h

```diff
@@ -3825,6 +3924,41 @@ TILEDB_EXPORT void tiledb_query_free(tiledb_query_t** query);
 TILEDB_EXPORT int32_t
 tiledb_query_submit(tiledb_ctx_t* ctx, tiledb_query_t* query);
 
+/**
+ * Submits a TileDB query after setting the provided subarray if query either
+ * previously completed or uninitialized.
+ *
+ * **Example:**
+ *
+ * @code{.c}
+ * tiledb_query_submit(ctx, query, subarray);
+ * @endcode
+ *
+ * @param ctx The TileDB context.
+ * @param query The query to be submitted.
+ * @param subarray The subarray to be used by the query.
+ * @return `TILEDB_OK` for success and `TILEDB_ERR` for error.
+ *
+ * @note `tiledb_query_finalize` must be invoked after finish writing in
+ *     global layout (via repeated invocations of `tiledb_query_submit`),
+ *     in order to flush any internal state.
+ *
+ * @note For the case of reads, if the returned status is `TILEDB_INCOMPLETE`,
+ *    TileDB could not fit the entire result in the user's buffers. In this
+ *    case, the user should consume the read results (if any), optionally
+ *    reset the buffers with `tiledb_query_set_buffer`, and then resubmit the
+ *    query until the status becomes `TILEDB_COMPLETED`. If all buffer sizes
+ *    after the termination of this function become 0, then this means that
+ *    **no** useful data was read into the buffers, implying that larger
+ *    buffers are needed for the query to proceed. In this case, the users
+ *    must reallocate their buffers (increasing their size), reset the buffers
+ *    with `tiledb_query_set_buffer`, and resubmit the query.
+ */
+TILEDB_EXPORT int32_t tiledb_query_submit_with_subarray(
```

Comment:
There doesn't seem to be any obvious reason for performing a tiledb_query_set_subarray_v2() except in preparation for a submit().
This api just attempts to directly express that otherwise implicit connection, and relieves a client from otherwise having to remember to do that.

Perhaps the separate tiledb_query_set_subarray_v2() routine isn't really needed publically...
maybe I'm overlooking something... ?

Reply:
The motivation is separation of concerns. Setting a subarray on the query is the same as setting buffers, modes, etc. Remove this API and use the `tiledb_query_set_subarray_v2` style API. Note that we actually want to call this `tiledb_query_set_subarray_t` instead of `_v2`. I've updated the ticket description to reflect this.